tests/smoke/smoke_test.go

```diff
@@ -0,0 +1,70 @@
+package smoke_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/accuknox/auto-policy-discovery/src/types"
+	"github.com/kubearmor/KubeArmor/tests/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rs/zerolog/log"
+)
+
+var _ = BeforeSuite(func() {
+	//install wordpress-mysql app
+	err := util.K8sApply([]string{"res/wordpress-mysql-deployment.yaml"})
+	Expect(err).To(BeNil())
+	time.Sleep(5 * time.Second)
+
+	// install wordpress-mysql app
+	_, err = util.Kubectl(fmt.Sprintf("apply -f https://raw.githubusercontent.com/kubearmor/discovery-engine/dev/deployments/k8s/deployment.yaml"))
+	Expect(err).To(BeNil())
+	time.Sleep(5 * time.Second)
+
+	// delete all KSPs
+	err = util.DeleteAllKsp()
+	Expect(err).To(BeNil())
+
+	// enable kubearmor port forwarding
+	err = util.KubearmorPortForward()
+	Expect(err).To(BeNil())
+})
+
+var _ = AfterSuite(func() {
+	util.KubearmorPortForwardStop()
+})
+
+func discover(ns string, l string) (types.KubeArmorPolicy, error) {
+	var policy types.KubeArmorPolicy
+	cmd, err := exec.Command("karmor", "discover", "-n", ns, "-l", l, "-f", "json").Output()
+	if err != nil {
+		log.Error().Msgf("Failed to apply the `karmor discover` command : %v", err)
+	}
+	err = json.Unmarshal(cmd, &policy)
+	if err != nil {
+		log.Error().Msgf("Failed to unmarshal the policy : %v", err)
+	}
+	return policy, err
+}
+
+var _ = Describe("Smoke", func() {
+
+	Describe("Auto Policy Discovery", func() {
+		It("test", func() {
+			policy, err := discover("wordpress-mysql", "app=wordpress")
+			Expect(err).To(BeNil())
+			Expect(policy.APIVersion).To(Equal("security.kubearmor.com/v1"))
+			Expect(policy.Kind).To(Equal("KubeArmorPolicy"))
+			Expect(policy.Metadata["namespace"]).To(Equal("wordpress-mysql"))
+			Expect(policy.Spec.Action).To(Equal("Allow"))
+			Expect(policy.Spec.Selector.MatchLabels["app"]).To(Equal("wordpress"))
+			Expect(policy.Spec.File.MatchDirectories[0].Dir).To(Equal("/usr/lib/x86_64-linux-gnu/"))
```

Comment:
expecting data to be at fixed position i.e. at index 0 in this case, may lead to an issue and key to flakyness.

Reply:
@rksharma95 what could be a possible solution for this, storing values in map with assigning a specific key, as these are slice?

Reply:
also can these values be changed on different host?

Reply:
> also can these values be changed on different host?

i don't think so, to be more confident we can target wordpress app specific policies they will not change.

> @rksharma95 what could be a possible solution for this, storing values in map with assigning a specific key, as MatchLabels is a map?

we can traverse slices i.e. MatchDirectories[ ], MatchPaths[ ] and look for the target values.

Reply:
> we can traverse slices i.e. MatchDirectories[ ], MatchPaths[ ] and look for the target values.

@rksharma95 can we do something like this
```
for i := range policy.Spec.File.MatchDirectories {
				if policy.Spec.File.MatchDirectories[i].Dir == "/etc/" {
					value = "/etc/"
				}
			}
			Expect(value).To(Equal("/etc/"))
```